src/ripple/app/ledger/impl/LedgerMaster.cpp

```diff
@@ -1181,12 +1180,7 @@ LedgerMaster::findNewLedgersToPublish(std::unique_lock<std::recursive_mutex>& sl
             "Gap in validated ledger stream " << mPubLedgerSeq <<
             " - " << mValidLedgerSeq - 1;
 
-        auto valLedger = mValidLedger.get ();
-        ret.push_back (valLedger);
-        setPubLedger (valLedger);
-        app_.getOrderBookDB().setup(valLedger);
```

Comment:
Why are we just removing this?

Reply:
@nbougalis This work is now done in `setFullLedger`. The conditional in that function was changed to handle this. The only way I could see this being an issue is the refactored code calls these functions later than the original code, but as far as I can tell the refactored code is correct.

Reply:
The only place this function, `findNewLedgersToPublish`, is called is [here](https://github.com/ripple/rippled/blob/2341710e91f2626001e5cd3abfda4d25e82e4ad1/src/ripple/app/ledger/impl/LedgerMaster.cpp#L1849) (all links are to the same commit as this PR) in `LedgerMaster::doAdvance()`. This line is returning a non-empty `pubLedgers` to `doAdvance()`, so it takes the else branch ([here](https://github.com/ripple/rippled/blob/2341710e91f2626001e5cd3abfda4d25e82e4ad1/src/ripple/app/ledger/impl/LedgerMaster.cpp#L1909)) of the if-statement [here](https://github.com/ripple/rippled/blob/2341710e91f2626001e5cd3abfda4d25e82e4ad1/src/ripple/app/ledger/impl/LedgerMaster.cpp#L1850).

In that else branch, we unconditionally call `setPubLedger(ledger)` ([here](https://github.com/ripple/rippled/blob/2341710e91f2626001e5cd3abfda4d25e82e4ad1/src/ripple/app/ledger/impl/LedgerMaster.cpp#L1923)) for the ledger we returned, which is the same as the call that used to be on line 1186.

For the call to `OrderBookDB::setup()` that was on line 1187, we must look into the call of `setFullLedger(ledger)` ([here](https://github.com/ripple/rippled/blob/2341710e91f2626001e5cd3abfda4d25e82e4ad1/src/ripple/app/ledger/impl/LedgerMaster.cpp#L1920)), which makes the same call [here](https://github.com/ripple/rippled/blob/2341710e91f2626001e5cd3abfda4d25e82e4ad1/src/ripple/app/ledger/impl/LedgerMaster.cpp#L856) (after no early returns).

Reply:
:+1: thanks both